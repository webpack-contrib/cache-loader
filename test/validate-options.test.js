import loader, { pitch } from '../src';

// Needed for `schema-utils` to not call
// process.exit(1) when running tests
process.env.JEST = true;

describe('validate options', () => {
  test('error', () => {
    const err = () => loader.call({ query: { cacheIdentifier: 1 } });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });

  test('unknown', () => {
    const err = () => loader.call({ query: { unknown: 'unknown' } });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });

  test('error (pitch)', () => {
    const err = () => pitch.call({ query: { cacheIdentifier: 1 } });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });

  test('unknown (pitch)', () => {
    const err = () => pitch.call({ query: { unknown: 'unknown' } });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });
});
