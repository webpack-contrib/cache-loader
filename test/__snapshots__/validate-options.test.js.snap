// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate options error (pitch) 1`] = `
"Invalid options object. Cache Loader (Pitch) has been initialised using an options object that does not match the API schema.
 - options.cacheIdentifier should be a string.
   -> Provide a cache directory where cache items should be stored (used for default read/write implementation)."
`;

exports[`validate options error 1`] = `
"Invalid options object. Cache Loader has been initialised using an options object that does not match the API schema.
 - options.cacheIdentifier should be a string.
   -> Provide a cache directory where cache items should be stored (used for default read/write implementation)."
`;

exports[`validate options unknown (pitch) 1`] = `
"Invalid options object. Cache Loader (Pitch) has been initialised using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { cacheContext?, cacheKey?, cacheIdentifier?, cacheDirectory?, compare?, precision?, read?, readOnly?, write? }"
`;

exports[`validate options unknown 1`] = `
"Invalid options object. Cache Loader has been initialised using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { cacheContext?, cacheKey?, cacheIdentifier?, cacheDirectory?, compare?, precision?, read?, readOnly?, write? }"
`;
